[RHACS] Release Notes for Version 3.72 (#50606)
openshift/openshift-docs: release_notes/372-release-notes.adoc

*Release date:* 26 September 2022 +

> {product-title} ({product-title-short}) is an enterprise-ready, Kubernetes-native container security solution that protects your vital applications across build, deploy, and runtime. It deploys in your infrastructure and integrates with your DevOps tooling and workflows to deliver better security and compliance and to enable DevOps and InfoSec teams to operationalize security.

Reviewer: `stages of application lifecycle`

[id="about-this-release-372"]
== About this release

{product-title-short} 3.72 includes:

> * Automatic removal of inactive clusters

Reviewer: `inactive` seems broad. We do not manage the removal of clusters that were added but not installed. We only manage removal of those clusters that were installed and did a handshake with Central at least once.
cc: @charmik-redhat

* Before this improvement, there were differences in the YAML file viewer instances. With this update, {product-title-short} shows the same formatting and interface when you view YAML files. 

[id="doc-updates-372"]
=== Documentation updates

The {product-title-short} documentation has been updated to include a xref:../operating/default-security-policies.adoc#default-security-policies[list of default security policies].

[id="notable-tech-changes-372"]
== Notable technical changes

[id="rhel9-scan"]
=== Scanning support for Red Hat Enterprise Linux 9

{product-title-short} 3.72 introduces support for analyzing images built with Red Hat Universal Base Image (UBI) 9 and Red Hat Enterprise Linux (RHEL) 9 RPMs for vulnerabilities.

RHEL 9 is now generally available (GA). RHEL 9 can run your code efficiently whether deployed on physical infrastructure, in a virtual machine, or in containers built from Red Hat UBIs. {product-title-short} supports RHEL 9 scanning with this release. 

[id="fixable-cvss6-disabled"]
=== Policy for CVEs with fixable CVSS of 6 or greater disabled by default 

> Beginning with this release, the `Fixable CVSS >= 6 and Privileged` policy is no longer enabled by default for new {product-title-short} installations. The configuration of this policy is not changed when upgrading an existing system. For new installations, this policy is replaced by the policy `Privileged Containers with Important and Critical Fixable CVEs`, which gives an alert for containers running in privileged mode that have important or critical fixable vulnerabilities. The new policy is also in the `Deploy` life cycle. Known vulnerabilities make it easier for adversaries to exploit your application, and highly-privileged containers pose greater risk. You can fix these high-severity vulnerabilities by updating to a newer version of the affected components or running your container with lower privileges.

Reviewer 2: ```suggestion
Beginning with this release, the `Fixable CVSS >= 6 and Privileged` policy is no longer enabled by default for new {product-title-short} installations. The configuration of this policy is not changed when upgrading an existing system. A new policy `Privileged Containers with Important and Critical Fixable CVEs`, which gives an alert for containers running in privileged mode that have important or critical fixable vulnerabilities, has now been added. The new policy is also in the `Deploy` life cycle. Known vulnerabilities make it easier for adversaries to exploit your application, and highly-privileged containers pose greater risk. You can fix these high-severity vulnerabilities by updating to a newer version of the affected components or running your container with lower privileges.
```

Maybe? To clarify that the new policy is added for both new and existing installations. Only the disabling is for new installations only.

** Retrieval by using the `props` fields will be removed in version a future release.
** Until removal, retrieval by using the `props` field will work if the result is unambiguous (no more than one group is found with the `props` field).
* `/v1/cves/suppress` and `/v1/cves/unsuppress` have been deprecated and will be removed in a future release. After these are removed:
** Use `/v1/imagecves/suppress` and `/v1/imagecves/unsuppress` to snooze and unsnooze image vulnerabilities.
** Use `/v1/nodecves/suppress` and `/v1/nodecves/unsuppress` to snooze and unsnooze node and host vulnerabilities.
** Use `/v1/clustercves/suppress` and `/v1/clustercves/unsuppress` to snooze and unsnooze platform (Kubernetes, Istio, and {ocp}) vulnerabilities.
* Permission `ClusterCVE` is deprecated and will be superseded by the existing permission `Cluster`. 
* `Label` and `Annotation` search options are deprecated and will be removed in the {product-title-short} 3.73 release. They will be replaced by the search options listed in the following table.

.Search options
[cols="1,1,1",options="header"]
|====
|Resource |Deprecated search option | New search option 

| Resource
| Deprecated Search Option

> | New Search Option

Reviewer: Twice

| Node
| Label
| Node Label

| Node
| Annotation
| Node Annotation

| Namespace
| Label
| Namespace Label

| Deployment
| Label
| Deployment Label

| ServiceAccount
| Label
| Service Account Label

| ServiceAccount
| Annotation
| Service Account Annotation

| K8sRole
| Label
| Role Binding Label

| K8sRoleAnnotation
| Annotation
| Role Binding Annotation

|====

[id="upcoming-api-changes"]
=== Notice of upcoming API changes

* CSV export: 

> ** Beginning in the {product-title-short} 3.73 release, the CSV export `API /api/vm/export/csv` will require the `CVE Type` filter as part of the input query parameter. Requests that do not have the filter will time out and give an error. Examples:

Reviewer 3: There won't be a timeout, just an error. So this can be updated to just say the requests will error out.

[id="view-dockerfile-line-cve"]
=== Ability to view Dockerfile lines in images that introduced components with Common Vulnerabilities and Exposures (CVEs)

In the *Images* view, under *Image Findings*, you can view individual lines in the Dockerfile that introduced the components that have been identified as containing CVEs. To view this information, locate the CVE from the list of CVEs provided in the *Image Findings* page. In the *Affected Components* column, click on the *<number> components* link. You can expand the display to show the line where the component was introduced that contains the CVE. For more information, see xref:../operating/examine-images-for-vulnerabilities.adoc#identify-dockerfile-line-component-cve_examine-images-for-vulnerabilities[Identifying the Dockerfile line in an image that adds a component with a CVE].

[id="network-graph-improvements"]
=== Network graph improvements

{product-title-short} 3.72 includes the following improvements to the Network Graph:

> * Before this improvement, the Legend in the Network Graph required hovering over the symbols representing namespaces, deployments, and connections. The updated legend shows the symbols and explanatory text together. 

Reviewer 4: Lowercase legend?

* Sometimes, the graph view edges did not connect the edges of nodes and overlapped on namespaces. This issue is fixed.
* Before this improvement, there were differences in the YAML file viewer instances. With this update, {product-title-short} shows the same formatting and interface when you view YAML files. 

[id="doc-updates-372"]
=== Documentation updates

The {product-title-short} documentation has been updated to include a xref:../operating/default-security-policies.adoc#default-security-policies[list of default security policies].

[id="notable-tech-changes-372"]
== Notable technical changes

[id="rhel9-scan"]
=== Scanning support for Red Hat Enterprise Linux 9

RHEL 9 is now generally available (GA). {product-title-short} 3.72 introduces support for analyzing images built with Red Hat Universal Base Image (UBI) 9 and Red Hat Enterprise Linux (RHEL) 9 RPMs for vulnerabilities.

[id="fixable-cvss6-disabled"]
=== Policy for CVEs with fixable CVSS of 6 or greater disabled by default 

> Beginning with this release, the `Fixable CVSS >= 6 and Privileged` policy is no longer enabled by default for new {product-title-short} installations. The configuration of this policy is not changed when upgrading an existing system. A new policy `Privileged Containers with Important and Critical Fixable CVEs`, which gives an alert for containers running in privileged mode that have important or critical fixable vulnerabilities, has been added. The new policy is also in the `Deploy` life cycle. Known vulnerabilities make it easier for adversaries to exploit your application, and highly-privileged containers pose greater risk. You can fix these high-severity vulnerabilities by updating to a newer version of the affected components or running your container with lower privileges.

Reviewer 4: IBM SG:

lifecycle
Write as shown: one word.

- TP: Technology Preview
- DEP: Deprecated
- REM: Removed

.Deprecated and removed features tracker
[cols="3,1,1,1",options="header"]
|====
|Feature | {product-title-short} 3.70 |{product-title-short} 3.71 | {product-title-short} 3.72

|`RenamePolicyCategory` and `DeletePolicyCategory` API endpoints
|DEP
|DEP
|DEP

|Support for violation tags and process tags
|DEP
|DEP
|REM

| Permissions: `AuthPlugin`, `AuthProvider`, `Group`, `Licenses`, `Role`, `User`, `Indicator`, `NetworkBaseline`, `ProcessWhitelist`, `Risk`, `APIToken`, `BackupPlugins`, `ImageIntegration`, `Notifier`, `SignatureIntegration`, `ImageComponent`
| GA
| DEP
| DEP

| Retrieving groups by property
| GA
| DEP
| DEP

| `vulns` fields of `storage.Node` object in response payload of `v1/nodes` 
| GA
| DEP
| DEP

| `/v1/cves/suppress` and `/v1/cves/unsuppress`
| GA
| DEP
| DEP

| `ids` field in the `/v1/cves/suppress` and `/v1/cves/unsuppress` API payload 
| GA
| DEP
| DEP

| `cves.ids` field of the `storage.VulnerabilityRequest` object in the response of `VulnerabilityRequestService` endpoints
| GA
| DEP
| DEP

| Scanning support for Ubuntu 21.10
| GA
| GA
| DEP

| Permission `ClusterCVE`
| GA
| GA
| DEP  

> | `Label` and `Annotation` search options

Reviewer 4: In the actual YAML, `label` and `annotation` are actually lowercase. This could just be _label_ and _annotation_ I think.

Author: added "in RHACS" to clarify that these are ACS search options

| GA
| GA
| DEP

|====

[id="deprecated-features-372"]
=== Deprecated features

* Permissions for permission sets will be grouped for simplification. The following list describes the new permissions and indicates the deprecated permissions that will be removed in a future release:
** The `Access` permission will replace the following permissions: `AuthPlugin`, `AuthProvider`, `Group`, `Licenses`, `Role`, and `User`.
** The `DeploymentExtension` permission will replace the following permissions: `Indicator`, `NetworkBaseline`, `ProcessWhitelist`, and `Risk`.
** The `Integration` permission will deprecate the following permissions: `APIToken`, `BackupPlugins`, `ImageIntegration`, `Notifier`, and `SignatureIntegration`.
** The `Image` permission will replace the permission `ImageComponent`.

> * Previously, groups were retrieved by using the field `props` (`props.authProviderId`, `props.key`, `props.value`). This field will be replaced by the new `props.id` field. Use the `props.id` field to retrieve groups in the Application Programming Interface (API). Note the following:

Reviewer 4: We generally avoid parenthesis when possible.

** Retrieval by using the `props` fields will be removed in version a future release.
** Until removal, retrieval by using the `props` field will work if the result is unambiguous (no more than one group is found with the `props` field).
* `/v1/cves/suppress` and `/v1/cves/unsuppress` have been deprecated and will be removed in a future release. After these are removed:
** Use `/v1/imagecves/suppress` and `/v1/imagecves/unsuppress` to snooze and unsnooze image vulnerabilities.
** Use `/v1/nodecves/suppress` and `/v1/nodecves/unsuppress` to snooze and unsnooze node and host vulnerabilities.
** Use `/v1/clustercves/suppress` and `/v1/clustercves/unsuppress` to snooze and unsnooze platform (Kubernetes, Istio, and {ocp}) vulnerabilities.
* Permission `ClusterCVE` is deprecated and will be superseded by the existing permission `Cluster`. 

> * `Label` and `Annotation` search options are deprecated and will be removed in the {product-title-short} 3.73 release. They will be replaced by the search options listed in the following table.

Reviewer 4: ```suggestion
* Label and annotation search options are deprecated and will be removed in the {product-title-short} 3.73 release. They will be replaced by the search options listed in the following table.
```

.Search options
[cols="1,1,1",options="header"]
|====

> |Resource |Deprecated search option | New search option 

Reviewer 4: These are actually Kubernetes API objects, such as `Node` or `Namespace` or `ServiceAccount`, so I'd probably put these in code font. But that's me.

Reviewer 2: As per the other comment this should be uppercase (and use ticks `) because they are referring to search terms within ACS which is uppercased.

| Node
| Label

> | Node Label

Reviewer 4: ```suggestion
| Node label
```

> | Node Label

Reviewer 4: I think these can be lowercase. This it the API object labels array. For example:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/name: myservice
    app.kubernetes.io/instance: myservice-abcxzy
```

Reviewer 2: As per the other comment this should be uppercase (and use ticks `) because they are referring to search terms within ACS which is uppercased.

| Node
| Annotation

> | Node Annotation

Reviewer 4: ```suggestion
| Node annotation
```

| Namespace
| Label
| Namespace Label

| Deployment
| Label
| Deployment Label

| ServiceAccount
| Label
| Service Account Label

| ServiceAccount
| Annotation
| Service Account Annotation

> | K8sRole

Reviewer 4: I'm not familiar with this API; I guess this is provided by RHACS?

| Label
| Role Binding Label

| K8sRoleAnnotation
| Annotation
| Role Binding Annotation

|====

[id="psp-changes"]
=== PodSecurityPolicy (PSP) objects and Kubernetes 1.25

The following changes were made in {product-title-short} to support the removal of PSP objects in Kubernetes 1.25:

> * PSP objects are created by default in installations for backward compatibility with Kubernetes versions prior to 1.25. However, you can manually disable PSP creation by setting `system.enablePodSecurityPolicies = false` in the Helm chart. 

Reviewer 4: Based on my read of Helm charts, I think this is best expressed as:

`system.enablePodSecurityPolicies: false`

* Beginning with {product-title-short} version 3.71, auto-sensing has been added to {product-title-short} Helm charts. If you install {product-title-short} using the Operator or Helm charts, and {product-title-short} detects a Kubernetes version of 1.25 or later, it will not install PSPs. If you are using the `roxctl` CLI to install {product-title-short}, you need to disable PSP usage by setting the `--enable-pod-security-policies` flag to `false` for the `roxctl central generate` and `roxctl sensor generate` commands.

Kubernetes users must disable the admission controller plugin for PSPs before upgrading to Kubernetes version 1.25. 

[id="upcoming-api-changes"]
=== Notice of upcoming API changes

> * CSV export: Beginning in the {product-title-short} 3.73 release, the CSV export `API /api/vm/export/csv` will require the `CVE Type` filter as part of the input query parameter. Requests that do not have the filter will give an error. Examples:

Reviewer 4: _return_ an error?

> ** `CVE Type:NODE_CVE`

Reviewer 4: ```suggestion
** `CVE_Type:NODE_CVE`
```

Reviewer 2: I believe this was done intentionally to show that you can use either of these options: `CVE Type`, `CVE_Type`, `CVE_TYPE`. @md2119 @charmik-redhat is that correct?

Reviewer: The correct option is `CVE Type`, ACS search label. It seems correctly captured above but has 3 variations in `Example` section. I recommend adding: 
```
Supported values for `CVE Type` are `IMAGE_CVE`, `K8S_CVE`, `ISTIO_CVE`, `NODE_CVE`, and `OPENSHIFT_CVE`.
```

** `CVE_Type:IMAGE_CVE`
** `CVE_TYPE:K8S_CVE`
* Suppress and unsuppress payloads: 
** The field `ids` in the `/v1/cves/suppress` and `/v1/cves/unsuppress` API payloads will be renamed to `cves` in the {product-title-short} 3.73 release.
** The `cves.ids` field of the `storage.VulnerabilityRequest` object in the response of `VulnerabilityRequestService` endpoints will be renamed to `cves.cves` in the {product-title-short} 3.73 release.

> [id="known-issues"]

Reviewer 4: ```suggestion
[id="known-issues-372"]
```

== Known issues

{product-title-short} shows the wrong severity when two severities exist for a single vulnerability in a single distribution. This issue occurs because {product-title-short} scopes severities by namespace rather than component. There is no workaround. (ROX-12527)

[id="bug-fixes-372"]
== Bug fixes

* Before this update, the steps to configure {ocp} OAuth for more than one URI were missing. The documentation has been revised to include instructions for configuring OAuth in {ocp} to use more than one URI. For more information, see xref:../operating/manage-user-access/configure-ocp-oauth.adoc#create-additional-routes-ocp-oauth_configure-ocp-oauth[Creating additional routes for the OpenShift Container Platform OAuth server]. (ROX-11296) 

> * Before this update, the autogenerated image integration (for example, a Docker registry integration) for a cluster is not deleted when the cluster is removed from Central. This issue is fixed. (ROX-9398)

Reviewer 4: We generally avoid parenthetical statements when possible per ISG.

> * Before this update, the autogenerated image integration (for example, a Docker registry integration) for a cluster is not deleted when the cluster is removed from Central. This issue is fixed. (ROX-9398)

Reviewer 4: Does a user know what _Central_ is in this context?

Author: I think it should be OK as is - it's one of the two major pieces of ACS and is usually just referred to as "Central" (no qualifiers).

> * Before this update, the `Image OS` policy criteria did not support regular expression, or `regex`. However, the documentation indicated that regular expression was supported. This issue is fixed by adding support for regular expression for the `Image OS` policy criteria. (ROX-12301)

Reviewer 4: ```suggestion
* Before this update, the `Image OS` policy criteria did not support regular expressions, or `regex`. However, the documentation indicated that regular expressions were supported. This issue is fixed by adding support for regular expressions for the `Image OS` policy criteria. (ROX-12301)
```

> * Before this update, the `syslog` integration did not respect a configured TCP proxy. This is now fixed.

Reviewer 4: syslog is so common, I think we actually just say _syslog_. Sort of like everyone writes ssh, rather than `ssh`, or bash rather than `bash`.

* Before this update, the `scanner-db` pod failed to start when a resource quota was set for the `stackrox` namespace, because the `init-db` container in the pod did not have any resources assigned to it. The `init-db` container for `ScannerDB` now specifies resource requests and limits that match the `db` container. (ROX-12291)

[id="image-versions-372"]
== Image versions

[%header,cols="1,1,2"]
|===
| Image | Description | Current version

| Main
| Includes Central, Sensor, Admission Controller, and Compliance.
Also includes `roxctl` for use in continuous integration (CI) systems.
| `registry.redhat.io/advanced-cluster-security/rhacs-main-rhel8:3.72`

| Scanner
| Scans images and nodes.
| `registry.redhat.io/advanced-cluster-security/rhacs-scanner-rhel8:3.72`

| Scanner DB
| Stores image scan results and vulnerability definitions.
| `registry.redhat.io/advanced-cluster-security/rhacs-scanner-db-rhel8:3.72`

| Collector
| Collects runtime activity in Kubernetes or {ocp} clusters.

> | `registry.redhat.io/advanced-cluster-security/rhacs-collector-rhel8:3.72`

Reviewer 4: If I expand the page big enough, these end up on a single line; Maybe use a bulleted list here?
